src/utils/xhr-loader.ts

```diff
@@ -164,37 +164,71 @@ class XhrLoader implements Loader<LoaderContext> {
         const status = xhr.status;
         // http status between 200 to 299 are all successful
         if (status >= 200 && status < 300) {
-          stats.loading.end = Math.max(
-            self.performance.now(),
-            stats.loading.first
-          );
-          let data;
-          let len: number;
-          if (context.responseType === 'arraybuffer') {
-            data = xhr.response;
-            len = data.byteLength;
-          } else {
-            data = xhr.responseText;
-            len = data.length;
-          }
-          stats.loaded = stats.total = len;
+          try {
+            stats.loading.end = Math.max(
+              self.performance.now(),
+              stats.loading.first
+            );
+            let data;
+            let len: number;
+            if (context.responseType === 'arraybuffer') {
```

Comment:
While `xhr.responseType` was set to `context.responseType` and should have the same value, we should only read off the xhr object here: 

```suggestion
            if (xhr.responseType === 'arraybuffer') {
```